csi/controller.go

```diff
@@ -1300,6 +1326,70 @@ func (s *OsdCsiServer) listMultipleSnapshots(
 	return listSnapshotsResp, nil
 }
 
+// ListSnapshots lists all cloud backups as snapshot in a cluster.
+func (s *OsdCsiServer) listCloudSnapshots(
+	ctx context.Context,
+	req *csi.ListSnapshotsRequest,
+) (*csi.ListSnapshotsResponse, error) {
+	cloudBackupClient, err := s.getCloudBackupClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	credentialID := req.GetSecrets()[osdSnapshotCredentialIDKey]
+
+	maxBackups := req.GetMaxEntries()
+	if len(req.GetSnapshotId()) > 0 {
+		maxBackups = 1
+	}
+	resp, err := cloudBackupClient.EnumerateWithFilters(ctx, &api.SdkCloudBackupEnumerateWithFiltersRequest{
+		SrcVolumeId:       req.GetSourceVolumeId(),
+		CloudBackupId:     req.GetSnapshotId(),
+		CredentialId:      credentialID,
+		ContinuationToken: req.GetStartingToken(),
+		MaxBackups:        uint64(maxBackups),
+	})
+	if nil != err {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to list cloud backups: %v", err)
+	}
+
+	csiSnapshotResp := &csi.ListSnapshotsResponse{
```

Comment:
Can you check if cloudBackupClient.EnumerateWithFilters returns a continuation token? Since we are passing in the ContinuationToken above, I'm guessing it does.

If so, we'll want to populate this `next_token` field in the response:
https://github.com/container-storage-interface/spec/blob/master/csi.proto#L1334